pygmt/helpers/utils.py

```diff
@@ -581,13 +581,13 @@ def build_arg_list(  # noqa: PLR0912
         gmt_args.extend(f"--{key}={value}" for key, value in confdict.items())
 
     if infile:  # infile can be a single file or a list of files
-        if isinstance(infile, str | pathlib.PurePath):
+        if isinstance(infile, str | os.PathLike):
             gmt_args = [str(infile), *gmt_args]
```

Comment:
> The GMT API does not support raw `bytes` as file paths. However, according to the Python documentation for `__fspath__`:
> 
> > The method should only return a str or bytest object, with a preference for str.
> 
> To my knowledge, no built-in Python classes return `bytes` from `__fspath__()`, so this case should be rare in practice. In earlier versions, such `bytes`-returning path-like objects are recognized as vectors; now, they are treated as file inputs instead.

Ok, let's assume the `bytes` case is rare enough to ignore. We're still doing `str(infile)` here, but maybe we'll need to use something like `str(infile.__fspath__())` instead if using `os.PathLike`?

Reply:
Changed to [`os.fspath`](https://docs.python.org/3/library/os.html#os.fspath) in commit 7c0a109fc1f874ab23afbe90fd63381ee9cf621f.
```
In [1]: from pathlib import Path

In [2]: import os

In [3]: os.fspath(Path("abc/def"))
Out[3]: 'abc/def'

In [4]: os.fspath("path/to/file")
Out[4]: 'path/to/file'
```

Reply:
Nice, I should have scrolled up the page a bit to see that `os.fspath` function! Also found [PEP 519](https://peps.python.org/pep-0519) to be a good reference that does seem to suggest `os.fspath`.